### PR TITLE
Add property to customize the display name for authenticated users

### PR DESCRIPTION
**Pull request by oplantalech**
Add new property `skin.user_display_name` which allows to display the email or the username in the log in button.

# Checks
- [ ] Runs on heroku
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [Giphy CAPTURE](https://giphy.com/apps/giphycapture) or [Peek](https://github.com/phw/peek)

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). It can help to figure out who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them either through slack or by assigning them as a reviewer on the PR

### CODE REVIEW

**Review by dippindots — COMMENTED** [review]
@oplantalech Thanks for making this pull request, this is a great feature! I put some comments below. I noticed that the core test are failing, it would be great if you can try to rebase to the latest master branch and update this pr again, thanks!

File: security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/keycloak/SAMLUserDetailsServiceImpl.java [hunk 1]
```diff
@@ -116,23 +125,23 @@ public Object loadUserBySAML(SAMLCredential credential)
         
         try {
             
-            if (userId == null) {
+            if (email == null) {
```
[the remark this answers is not in the document]
Comment (oplantalech) [reply security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/keycloak/SAMLUserDetailsServiceImpl.java:129]: Currently we're only working with the email (as `userId`), so the idea of the new implementation is that if there is no `userName`, the app works with `email` (just as now). Therefore, no error should be thrown if `userName` is not defined.

**Comment by dippindots** [comment]
I still see the error after rebase, I believe it's related to the parameters here: https://github.com/cBioPortal/cbioportal/blob/master/portal/src/integration-tests/saml-oauth2-setup/pom.xml

**Comment by oplantalech** [comment]
@dippindots Addressed your comments, and I've added a small piece in the documentation, since an extra mapper in Keycloak needs to be created to pass the _username_ property.

**Comment by oplantalech** [comment]
@dippindots @alisman  Important: the Frontend PR (https://github.com/cBioPortal/cbioportal-frontend/pull/4498) needs to be merged first

**Comment by oplantalech** [comment]
@dippindots Well we still need to merge the Frontend PR first. You might want to solve the core issue first :)

**Comment by oplantalech** [comment]
@dippindots Well we still need to merge the Frontend PR first. You might want to solve the core issue first :)

**Comment by dippindots** [comment]
@oplantalech I'm confused... You mean after we merge the frontend, the core test build issue will be fixed?

**Comment by oplantalech** [comment]
@dippindots I mean: this PR needs to wait for https://github.com/cBioPortal/cbioportal-frontend/pull/4498 to be merged. So you might want to separate fixing the core tests in a different PR than this one here

**Comment by dippindots** [comment]
@oplantalech I see, thanks for clarifying this. However, the core tests are passing in the master branch, see here: https://github.com/cBioPortal/cbioportal/actions/runs/4117931797/jobs/7109838595.

So the build error you see in the core test is introduced in this pr, that's why I was trying to see if you can fix it in this pr.

**Comment by dippindots** [comment]
@oplantalech I'm not an expert in this field, but I guess it's related to some parameters you changes that need to be updated in this file: parameters here: https://github.com/cBioPortal/cbioportal/blob/master/portal/src/integration-tests/saml-oauth2-setup/pom.xml.

Just my two cents.

**Comment by oplantalech** [comment]
@dippindots Sorry, then I did not understand you. I will try to figure out what goes wrong in the tests then

**Comment by pvannierop** [comment]
[Add property to customize the display name for authenticated users](https://trello.com/c/oEMAsFXH/6-add-property-to-customize-the-display-name-for-authenticated-users)

**Comment by oplantalech** [comment]
@dippindots I've fixed the core tests

**Comment by dippindots** [comment]
@oplantalech That's awesome! Thank you! I will merge this into a branch under the cBioPortal account and let it run related CircleCi tests before I merge it to the master branch.

**Pull request merged**